### DISCUSSION TITLE
Use memo with next-intl

### DISCUSSION THREAD

**Discussion by Impesud** [amannn/next-intl#260] [Q&A]
Hi all, how (if any) are you using Memo with next-intl?

```
const Header: FC = memo(() => {
   const t = useTranslations('Header');
   const [currentSection, setCurrentSection] = useState<SectionId | null>(null);
   const navSections = useMemo(
     () => [t(SectionId.About), t(SectionId.Resume), t(SectionId.Portfolio), t(SectionId.Testimonials), t(SectionId.Contact)],
     [],
   );
```

Clearly this doesn't work properly because then it doesn't change the value. Do you kindly have any ideas?

**Comment by amannn** [comment]
Hey! Did you run into a performance problem that is related to `next-intl`? The `t` function is already memoized internally (both the function reference, as well as calls to it), so there should be no need to add further memoization. 

Hope this helps!
